### PR TITLE
MGMT-8958: Proxy support in ACI

### PR DESCRIPTION
**Pull request by rollandf**
Add Proxy configuration support in AgentClusterInstall
Signed-off-by: Fred Rolland <frolland@redhat.com>

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @filanov 
/cc @nmagnezi 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/controllers/clusterdeployments_controller.go [hunk 1]
```diff
@@ -875,6 +877,25 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context,
 	return clusterAfterUpdate, nil
 }
 
+func updateProxyParams(proxy *hiveext.Proxy, params *models.V2ClusterUpdateParams) bool {
+	update := false
+	if proxy != nil {
+		if proxy.NoProxy != "" {
```
Comment (filanov) [inline internal/controller/controllers/clusterdeployments_controller.go:883]: should probably be similar to https://github.com/openshift/assisted-service/blob/202ce128d8a31fed5cf1bcfeeb2b3f14176ddab2/internal/controller/controllers/clusterdeployments_controller.go#L740-L745 
if user delete the configuration we should update that as well no? or at least set it to empty string.

Comment (rollandf) [reply internal/controller/controllers/clusterdeployments_controller.go]: Updated.
I believe that there a lot of similar issues in the controllers when the user delete/clear configurations.
Worth to check.

Comment (filanov) [reply internal/controller/controllers/clusterdeployments_controller.go]: I saw that, will try to look at it thanks.

**Comment by eliorerz** [comment]
/retitle Bug 2049876: Proxy support in ACI

**Pull request merged**